### PR TITLE
LibWeb: Compute default ARIA roles context-sensitively where required

### PR DESCRIPTION
**Pull request by sideshowbarker**
This change implements spec-conformant computation of default ARIA roles for elements whose expected default role depends on the element’s context — specifically, either on the element’s ancestry, or on whether the element has an accessible name, or both. This affects the `aside`, `footer`, `header`, and `section` elements.

Otherwise, without this change, `aside`, `footer`, `header`, and `section` elements may unexpectedly end up with the wrong default roles.

This gets us passing 100% of the subtests in the `roles-contextual.html` test at https://wpt.fyi/results/html-aam?product=ladybird.

This change includes an additional commit that adds a `window.internals.getComputedLabel(element)` function, for use in our imported copies of the WPT tests related to the behavior in this change.

### CODE REVIEW

File: Libraries/LibWeb/HTML/HTMLElement.cpp [hunk 1]
```diff
@@ -732,8 +732,19 @@ Optional<ARIA::Role> HTMLElement::default_role() const
...
         return ARIA::Role::article;
     // https://www.w3.org/TR/html-aria/#el-aside
-    if (local_name() == TagNames::aside)
+    if (local_name() == TagNames::aside) {
+        // https://w3c.github.io/html-aam/#el-aside
+        for (auto const* ancestor = parent_element(); ancestor; ancestor = ancestor->parent_element()) {
+            if ((ancestor->local_name() == TagNames::article
+                    || ancestor->local_name() == TagNames::aside
+                    || ancestor->local_name() == TagNames::nav
+                    || ancestor->local_name() == TagNames::section)
```
Comment (tcl3) [inline Libraries/LibWeb/HTML/HTMLElement.cpp:741]: You can make this a bit more succinct with `first_is_one_of(ancestor->local_name(), TagNames::article, TagNames::aside, TagNames::main, TagNames::nav, TagNames::section)`.

File: Libraries/LibWeb/HTML/HTMLElement.cpp [hunk 2]
```diff
@@ -756,16 +767,31 @@ Optional<ARIA::Role> HTMLElement::default_role() const
...
+                || ancestor->local_name() == TagNames::aside
+                || ancestor->local_name() == TagNames::main
+                || ancestor->local_name() == TagNames::nav
+                || ancestor->local_name() == TagNames::section)
+                return ARIA::Role::generic;
+            if (ancestor_role == ARIA::Role::article
+                || ancestor_role == ARIA::Role::complementary
+                || ancestor_role == ARIA::Role::main
+                || ancestor_role == ARIA::Role::navigation
+                || ancestor_role == ARIA::Role::region)
```
Comment (tcl3) [inline Libraries/LibWeb/HTML/HTMLElement.cpp:787]: These could also be simplified with `first_is_one_of()`.

File: Libraries/LibWeb/HTML/HTMLElement.cpp (hunk 1 above)
Comment (sideshowbarker) [reply Libraries/LibWeb/HTML/HTMLElement.cpp:741]: > You can make this a bit more succinct with `first_is_one_of(ancestor->local_name(), TagNames::article, TagNames::aside, TagNames::main, TagNames::nav, TagNames::section)`.

Thanks — made it so

Comment (tcl3) [reply Libraries/LibWeb/HTML/HTMLElement.cpp:741]: This doesn't appear to have changed. Is there a change that needs to be pushed?

Comment (sideshowbarker) [reply Libraries/LibWeb/HTML/HTMLElement.cpp]: oops, thanks — there were two instances that needed replacement with `first_is_one_of`, but I had only replaced one

**Review by tcl3 — APPROVED** [review]

**Pull request merged**